app/components/InputStyler/InputStyler.tsx

```diff
@@ -1,12 +1,70 @@
-import { Tag, Input, Radio, AutoComplete, InputNumber, Flex } from "antd";
+import { Tag, Input, Radio, AutoComplete, InputNumber, Flex, Button } from "antd";
 import { Scenario } from "@/app/types/scenario";
+interface TemplateObject {
+  [key: string]: any;
+}
+
+interface ParsedSchema {
+  arrayName: string;
+  objectTemplate: TemplateObject;
+}
+
+interface rawDataProps {
+  [key: string]: any;
+  rulemap?: boolean;
+}
+
+export const parseSchemaTemplate = (template: string): ParsedSchema | null => {
+  if (!template) return null;
+  const match = template.match(/(\w+)\[\{(.*)\}\]/);
+  if (!match) {
+    return null;
+  }
+
+  const arrayName = match[1];
+  const properties = match[2].split(",").map((prop) => prop.trim());
+
+  const objectTemplate: TemplateObject = {};
+  properties.forEach((prop) => {
+    const [propertyName, propertyType] = prop.split(":");
+    switch (propertyType.toLowerCase()) {
+      case "string":
+        objectTemplate[propertyName] = "";
+        break;
+      case "boolean":
+        objectTemplate[propertyName] = false;
+        break;
+      case "number":
+        objectTemplate[propertyName] = 0;
+        break;
+      default:
+        objectTemplate[propertyName] = undefined;
+        break;
+    }
+  });
+
+  return { arrayName, objectTemplate };
+};
+
+const generateArrayFromSchema = (template: string, initialSize: number = 1): TemplateObject[] | null => {
+  if (!template || typeof template !== "string") return null;
+  const { objectTemplate } = parseSchemaTemplate(template) ?? {};
+  if (!objectTemplate) return null;
+
+  const array: TemplateObject[] = [];
+  for (let i = 0; i < initialSize; i++) {
+    array.push({ ...objectTemplate });
+  }
+
+  return array;
+};
 
-export default function inputStyler(
+export default function InputStyler(
```

Comment:
Might want to break this file up a bit or at least rename it because it feels like it's doing a lot more than just styling an input.

Reply:
Refactored array logic into its own component.